src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer_Diagnostics.cs

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using DiagnosticIds = Roslyn.Diagnostics.Analyzers.RoslynDiagnosticIds;
+
+namespace Microsoft.CodeAnalysis.PublicApiAnalyzers
+{
+    using static PublicApiAnalyzerResources;
+
+    public partial class DeclarePublicApiAnalyzer
+    {
+        internal static readonly DiagnosticDescriptor DeclareNewPublicApiRule = new(
+            id: DiagnosticIds.DeclarePublicApiRuleId,
+            title: CreateLocalizableResourceString(nameof(DeclarePublicApiTitle)),
+            messageFormat: CreateLocalizableResourceString(nameof(DeclarePublicApiMessage)),
+            category: "ApiDesign",
+            defaultSeverity: DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            description: CreateLocalizableResourceString(nameof(DeclarePublicApiDescription)),
+            helpLinkUri: "https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md",
+            customTags: WellKnownDiagnosticTagsExtensions.Telemetry);
+
+        internal static readonly DiagnosticDescriptor DeclareNewInternalApiRule = new(
+            id: DiagnosticIds.DeclareInternalApiRuleId,
```

Comment:
Was there a specific need for using separate diagnostic IDs for the internal and public API rules? If the only difference is in the diagnostic messages, we can potentially have both the descriptors use the same ID and rest of the fields, but use different messages.

Reply:
Yes. The intent was to have internal tracking disabled by default, and turned on by enabling the diagnostics.